#### Data Exploration: Investigate Data

In [2]:
%run './child/libraries.ipynb'

In [3]:
import pandas as pd
import random
import shelve

In [9]:
shared1 = load_dict('shared1')
dfFlt  = load_df('dfFlt')

In [10]:
list_todo()

{'todo': 'Investigate 3298 null values in response_correctness', 'finished': True}
{'todo': 'Investigate 1566 [unassigned] in response_correctness', 'finished': True}
{'todo': 'Investigate null values in item_type_code_name', 'finished': False}
{'todo': 'Reduce number of levels in item_type_code_name', 'finished': False}
{'todo': 'Remove unary learner_attempt_status = "fully scored"', 'finished': True}
{'todo': 'Investigate number_of_learners > 40 outliers', 'finished': False}
{'todo': 'Investigate binary variables assignment_attempt_number and assignment_max_attempts', 'finished': True}
{'todo': 'Investigate assignment_attempt_number and assignment_max_attempts both have 1566 values', 'finished': True}
{'todo': 'Investigate final_score_unweighted has 16,582 zero values', 'finished': False}
{'todo': 'Replace default dates with nulls', 'finished': False}
{'todo': 'Investigate why some dates are null', 'finished': False}
{'todo': 'Remove variables assignment_late_submission, is_manual_sc

###### Create Dataframe for Investigating Data

In [11]:
dfInv1 = dfFlt
shared2 = shared1

###### Remove variables assignment_late_submission, is_manual_scoring_required and is_deleted
Do this early so they don't get in the way

In [12]:
dfInv1 = dfInv1.drop('assignment_late_submission', 'is_manual_scoring_required', 'is_deleted')
shared2['binaryVars'].remove('assignment_late_submission')
shared2['binaryVars'].remove('is_manual_scoring_required')
shared2['binaryVars'].remove('is_deleted')

In [13]:
finish_todo('Remove variables assignment_late_submission, is_manual_scoring_required and is_deleted')

###### Remove unary learner_attempt_status
Do this early so they don't get in the way

In [14]:
dfInv1 = dfInv1.drop('learner_attempt_status')
shared2['nominalVars'].remove('learner_attempt_status')

In [18]:
save_df(dfInv1, 'dfInv1')
save_dict(shared2, 'shared2')

In [19]:
finish_todo('Remove unary learner_attempt_status = "fully scored"')

- [attempt-number-vs-max](./investigation/attempt-number-vs-max.ipynb)


- [1566-unassigned-in-response-correctness](./investigation/1566-unassigned-in-response-correctness.ipynb)


In [51]:
dfInv2 = load_df('dfInv2')



- [3298-null-in-response-correctness](./investigation/3298-null-in-response-correctness.ipynb)


In [5]:
dfInv3 = load_df('dfInv3')
shared = load_dict('shared')

In [6]:
list_todo()

{'todo': 'Investigate 3298 null values in response_correctness', 'finished': True}
{'todo': 'Investigate 1566 [unassigned] in response_correctness', 'finished': True}
{'todo': 'Investigate null values in item_type_code_name', 'finished': False}
{'todo': 'Reduce number of levels in item_type_code_name', 'finished': False}
{'todo': 'Remove unary learner_attempt_status = "fully scored"', 'finished': True}
{'todo': 'Investigate number_of_learners > 40 outliers', 'finished': False}
{'todo': 'Investigate binary variables assignment_attempt_number and assignment_max_attempts', 'finished': True}
{'todo': 'Investigate assignment_attempt_number and assignment_max_attempts both have 1566 values', 'finished': True}
{'todo': 'Investigate final_score_unweighted has 16,582 zero values', 'finished': False}
{'todo': 'Replace default dates with nulls', 'finished': False}
{'todo': 'Investigate why some dates are null', 'finished': False}
{'todo': 'Remove variables assignment_late_submission, is_manual_sc

- [item-type-code-name-nulls](./investigation/item-type-code-name-nulls.ipynb)


###### Statistics of final_score_unweighted to item_type_code_name = Null

In [ ]:
dfInv.filter(F.col("item_type_code_name").isNull() == True).select("final_score_unweighted").toPandas().hist(bins=60)

- Strongly correlated to a final_score_unweighted of 0

In [ ]:
finish_todo('Investigate null values in item_type_code_name')


